source/factory.js

```diff
@@ -1,35 +1,39 @@
+'use strict';
+
+/* eslint-disable consistent-return, no-param-reassign */
```

Comment:
I enabled `eslint-disable` because I literally don’t understand what this module does and how to change it to satisfy the ESLint rules.